File: app-android/src/main/java/com/ustadmobile/port/android/view/EntityNodeBinder.java
package com.ustadmobile.port.android.view;

import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import com.toughra.ustadmobile.R;
import com.ustadmobile.core.controller.CommonEntityHandlerPresenter;

import tellh.com.recyclertreeview_lib.TreeNode;
import tellh.com.recyclertreeview_lib.TreeViewBinder;

public class EntityNodeBinder extends TreeViewBinder<EntityNodeBinder.ViewHolder> {

    CommonEntityHandlerPresenter mPresenter;

    EntityNodeBinder(CommonEntityHandlerPresenter presenter){
        this.mPresenter = presenter;
    }

    @Override
    public ViewHolder provideViewHolder(View view) {
        return new ViewHolder(view);
    }

    @Override
    public void bindView(ViewHolder viewHolder, int i, TreeNode treeNode) {
        EntityLayoutType locationNode = (EntityLayoutType) treeNode.getContent();
        viewHolder.tvName.setText(locationNode.name);
        viewHolder.locationUid = locationNode.uid;
    }

    @Override
    public int getLayoutId() {
        return R.layout.item_select_multiple_tree_dialog;
    }

    public class ViewHolder extends TreeViewBinder.ViewHolder{

        private ImageView ivArrow;
        TextView tvName;
        CheckBox checkBox;
        Long locationUid;


        public ImageView getIvArrow() {
            return ivArrow;
        }

        public CheckBox getCheckBox() {
            return checkBox;
        }

        public ViewHolder(View rootView) {
            super(rootView);

            this.tvName = rootView.findViewById(R.id.item_select_multiple_tree_dialog_name);
            this.ivArrow = rootView.findViewById(R.id.item_select_multiple_tree_dialog_arrow);
            this.checkBox = rootView.findViewById(R.id.item_select_multiple_tree_dialog_checkbox);

            checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
                this.checkBox.setChecked(isChecked);
                mPresenter.entityChecked(tvName.getText().toString(), locationUid, isChecked);

            });
        }
    }
}

File: app-android/src/main/kotlin/com/ustadmobile/port/android/view/EntityNodeBinder.kt
package com.ustadmobile.port.android.view

import android.view.View
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView

import com.toughra.ustadmobile.R
import com.ustadmobile.core.controller.CommonEntityHandlerPresenter

import tellh.com.recyclertreeview_lib.TreeNode
import tellh.com.recyclertreeview_lib.TreeViewBinder

class EntityNodeBinder (
        var mPresenter: CommonEntityHandlerPresenter<*>)
    : TreeViewBinder<EntityNodeBinder.TreeHolder>() {

    override fun getLayoutId(): Int {
        return R.layout.item_select_multiple_tree_dialog
    }

    override fun provideViewHolder(view: View): TreeHolder {
        return TreeHolder(view)
    }

    override fun bindView(viewHolder: TreeHolder, i: Int, treeNode: TreeNode<*>?) {
        val locationNode = treeNode!!.getContent() as EntityLayoutType
        viewHolder.tvName.text = locationNode.name
        viewHolder.locationUid = locationNode.uid
    }

    inner class TreeHolder(rootView: View) : TreeViewBinder.ViewHolder(rootView) {

        val ivArrow: ImageView
        internal var tvName: TextView
        var checkBox: CheckBox
            internal set
        internal var locationUid: Long? = null

        init {
            this.tvName = rootView.findViewById(R.id.item_select_multiple_tree_dialog_name)
            this.ivArrow = rootView.findViewById(R.id.item_select_multiple_tree_dialog_arrow)
            this.checkBox = rootView.findViewById(R.id.item_select_multiple_tree_dialog_checkbox)

            checkBox.setOnCheckedChangeListener { buttonView, isChecked ->
                this.checkBox.isChecked = isChecked
                mPresenter.entityChecked(tvName.text.toString(), locationUid, isChecked)

            }
        }
    }
}